share/docker/openfast_ubuntu/Dockerfile

```diff
@@ -13,42 +12,63 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 #
+ARG BASE=ubuntu:jammy
 
-FROM ubuntu:bionic
+# Build stage 1: builds openfast.
+FROM ${BASE} AS build
 
 # Install dependencies
-
 # For gfortran-8
 # RUN add-apt-repository ppa:ubuntu-toolchain-r/test -y
 # apt-get install gfortran-8
 
-ENV DEBIAN_FRONTEND=noninteractive TZ=America/Denver
-
-RUN apt update -qq && \
-    apt install -y software-properties-common build-essential && \
-    add-apt-repository ppa:git-core/ppa -y && \
-    apt install -y python3-pip && \
-    apt install -y cmake cmake-curses-gui && \
-    apt install -y gcc gfortran make && \
-    apt install -y libblas-dev liblapack-dev && \
-    apt install -y git && \
-    apt install -y nano
+ENV DEBIAN_FRONTEND=noninteractive
 
-RUN pip3 install numpy
+RUN apt-get update -qq && \
+    apt-get install --no-install-recommends -y \
+        software-properties-common \
+        build-essential \
+        cmake  \
+        cmake-curses-gui \
+        gcc \
+        gfortran \
+        git \
+        libblas-dev \
+        liblapack-dev \
+        make \
+    && rm -rf /var/lib/apt/lists/*
 
-# Configure the environment
-ENV FC=/usr/bin/gfortran
-
-# Clone the project
-RUN git clone --recursive https://github.com/openfast/openfast.git openfast
+# Copy in the checked-out code version.
 WORKDIR /openfast
+COPY . .
 
-# Build the project
+# Build the project.
 RUN mkdir build
 WORKDIR /openfast/build
 
 # NOTE: building with optimizations on (RELEASE or RELWITHDEBINFO), the virtual machine
-# will require about 6GB of memoery. Otherwise, the gfortran compiler will exit with an
+# will require about 6GB of memory. Otherwise, the gfortran compiler will exit with an
 # "internal error"
-RUN cmake .. -DBUILD_TESTING=ON -DDOUBLE_PRECISION=ON -DCMAKE_BUILD_TYPE=RELWITHDEBINFO
-RUN make -j4 install
+ENV FC=/usr/bin/gfortran
+ARG CMAKE_OPTIONS="-DBUILD_TESTING=OFF -DDOUBLE_PRECISION=ON -DCMAKE_BUILD_TYPE=RELEASE"
```

Comment:
It might be worthwhile to go ahead and add FAST.Farm, though then we might need to pull in the OpenMP libraries.

Reply:
After chatting with @deslaughter, I think it makes sense to put FAST.Farm in this container as well.  It wouldn't have OpenMP so it would be of limited value for production use, but could be very useful for troubleshooting a setup before sending it to a cluster to run (FAST.Farm really should be compiled with Intel for production usage using native libraries on the cluster).